src/utils/util.ts

```diff
@@ -72,6 +74,52 @@ ${docs.tags
   .join('\n')}`.trim();
 }
 
+/**
+ * Adds a doc block to a string
+ * @param str the string to add a doc block to
+ * @param docs the compiled JS docs
+ * @param indentation number of spaces to indent the block with
+ * @returns the doc block
+ */
+export function addDocBlock(str: string, docs?: d.CompilerJsDoc, indentation: number = 0): string {
+  if (!docs) {
+    return str;
+  }
+
+  return [formatDocBlock(docs, indentation), str].filter(Boolean).join(`\n`);
+}
+
+/**
+ * Formats the given compiled docs to a JavaScript doc block
+ * @param docs the compiled JS docs
+ * @param indentation number of spaces to indent the block with
+ * @returns the formatted doc block
+ */
+function formatDocBlock(docs: d.CompilerJsDoc, indentation: number = 0): string {
+  const textDocs = getDocBlockLines(docs);
+  if (!textDocs.filter(Boolean).length) {
+    return '';
+  }
+
+  const spaces = new Array(indentation + 1).join(' ');
+
+  return [spaces + '/**', ...textDocs.map((line) => spaces + ` * ${line}`), spaces + ' */'].join(`\n`);
+}
+
+/**
+ * Get all lines part of the doc block
+ * @param docs the compiled JS docs
+ * @returns list of lines part of the doc block
+ */
+function getDocBlockLines(docs: d.CompilerJsDoc): string[] {
+  return [
+    ...docs.text.split(lineBreakRegex),
+    ...docs.tags
+      .filter((tag) => !SUPPRESSED_JSDOC_TAGS.includes(tag.name))
+      .map((tag) => `@${tag.name} ${tag.text || ''}`.split(lineBreakRegex)),
+  ].flat();
```

Comment:
I noticed something after a bit more testing - if a component's JSDoc comment doesn't start with a few lines of prose, and instead starts with a JSDoc tag like `@virtualProp`, `@slot`, etc, like so:

```ts
/**
 * @Prop first a string!
 */
@Component({
  tag: 'my-component',
  styleUrl: 'my-component.css',
  shadow: true,
})
export class MyComponent {
```

then we end up with a JSDoc comment in `components.d.ts` that looks like this:

```ts
    /**
     * 
     * @Prop first a string!
     */
    interface MyComponent {
```

I did a little testing and I think if we make the following change (or an equivalent one) here:

```suggestion
  ].flat().filter(line => line !== "");
```

we can eliminate the extra blank line, to get something like:

```ts
    /**
     * @Prop first a string!
     */
    interface MyComponent {
```

Reply:
909baf3